File: Algorithms/Implementation/Viral Advertising.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

class Result
{

    /*
     * Complete the 'viralAdvertising' function below.
     *
     * The function is expected to return an INTEGER.
     * The function accepts INTEGER n as parameter.
     */

    public static int viralAdvertising(int n)
    {

        int shared = 5;
        int liked = 0;
        int cumulative = 0;
        for (int i = 0; i < n; i++)
        {
            liked = shared / 2;
            cumulative += liked;
            shared = liked * 3;
        }
        return cumulative;
    }
}

class Solution
{
    public static void Main(string[] args)
    {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        int n = Convert.ToInt32(Console.ReadLine().Trim());
        int result = Result.viralAdvertising(n);

        textWriter.WriteLine(result);

        textWriter.Flush();
        textWriter.Close();
    }
}

File: Algorithms/Implementation/Viral Advertising.php
<?php

/*
 * Complete the 'viralAdvertising' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts INTEGER n as parameter.
 */

function viralAdvertising($n) {
    # Write your code here
    $shared = 5;
    $cumulative = 0;
    for ($i = 0; $i < $n; $i++) {
        $liked = intval($shared / 2);
        $cumulative += $liked;
        $shared = $liked * 3;
    }
    return $cumulative;
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$n = intval(trim(fgets(STDIN)));
$result = viralAdvertising($n);

fwrite($fptr, $result . "\n");
fclose($fptr);
